packages/block-editor/src/components/inserter/reusable-block-list.js

```diff
@@ -0,0 +1,111 @@
+/**
+ * External dependencies
+ */
+import { isEmpty } from 'lodash';
+
+/**
+ * WordPress dependencies
+ */
+import { withSpokenMessages } from '@wordpress/components';
+import { useMemo, useEffect } from '@wordpress/element';
+import { __, _n, sprintf } from '@wordpress/i18n';
+import { addQueryArgs } from '@wordpress/url';
+
+/**
+ * Internal dependencies
+ */
+import BlockTypesList from '../block-types-list';
+import __experimentalInserterMenuExtension from '../inserter-menu-extension';
+import { searchBlockItems } from './search-items';
+import InserterPanel from './panel';
+import InserterNoResults from './no-results';
+import useBlockTypesState from './hooks/use-block-types-state';
+
+/**
+ * List of reusable blocks shown in the "Reusable" tab of the inserter.
+ *
+ * @param {Object}   props                Component props.
+ * @param {?string}  props.rootClientId   Client id of block to insert into.
+ * @param {Function} props.onInsert       Callback to run when item is inserted.
+ * @param {Function} props.onHover        Callback to run when item is hovered.
+ * @param {?string}  props.filterValue    Search term.
+ * @param {Function} props.debouncedSpeak Debounced speak function.
+ *
+ * @return {WPComponent} The component.
+ */
+export function InserterReusableBlockList( {
+	rootClientId,
+	onInsert,
+	onHover,
+	filterValue,
+	debouncedSpeak,
+} ) {
+	const [ items, categories, collections, onSelectItem ] = useBlockTypesState(
+		rootClientId,
+		onInsert
+	);
+
+	const filteredItems = useMemo( () => {
+		return searchBlockItems(
+			items,
+			categories,
+			collections,
+			filterValue
+		).filter( ( { category } ) => category === 'reusable' );
```

Comment:
Not for this PR, but I think we should let go of "reusable" being a category and instead allow the tab to display other categories. That would allow a site to categorize reusable blocks (with the regular category interface when managing the custom post type) and have the categories listed in this tab as regular block categories.

Reply:
Agreed.